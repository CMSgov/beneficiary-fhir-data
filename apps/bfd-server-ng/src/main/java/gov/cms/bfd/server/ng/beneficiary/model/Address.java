package gov.cms.bfd.server.ng.beneficiary.model;

import jakarta.persistence.Column;
import jakarta.persistence.Embeddable;
import java.util.Optional;

@Embeddable
public class Address {
  @Column(name = "geo_usps_state_cd")
  private String stateCode;

  @Column(name = "geo_zip5_cd")
  private String zipCode;

  @Column(name = "geo_zip_plc_name")
  private String city;

  @Column(name = "bene_line_1_adr")
  private Optional<String> addressLine1;

  @Column(name = "bene_line_2_adr")
  private Optional<String> addressLine2;

  @Column(name = "bene_line_3_adr")
  private Optional<String> addressLine3;

  @Column(name = "bene_line_4_adr")
  private Optional<String> addressLine4;

  @Column(name = "bene_line_5_adr")
  private Optional<String> addressLine5;

  @Column(name = "bene_line_6_adr")
  private Optional<String> addressLine6;

  org.hl7.fhir.r4.model.Address toFhir() {
    var address =
        new org.hl7.fhir.r4.model.Address()
            .setState(stateCode)
            .setPostalCode(zipCode)
            .setCity(city);
    addressLine1.ifPresent(address::addLine);
    addressLine2.ifPresent(address::addLine);
    addressLine3.ifPresent(address::addLine);
    addressLine4.ifPresent(address::addLine);
    addressLine5.ifPresent(address::addLine);
    addressLine6.ifPresent(address::addLine);

    return address;
  }
}
