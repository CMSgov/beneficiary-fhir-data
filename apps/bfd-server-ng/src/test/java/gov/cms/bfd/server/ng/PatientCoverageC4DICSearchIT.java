package gov.cms.bfd.server.ng;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertTrue;
import static org.junit.jupiter.api.Assertions.fail;

import ca.uhn.fhir.rest.api.SearchStyleEnum;
import gov.cms.bfd.server.ng.beneficiary.BeneficiaryRepository;
import gov.cms.bfd.server.ng.util.SystemUrls;
import org.hl7.fhir.r4.model.Bundle;
import org.hl7.fhir.r4.model.Coverage;
import org.hl7.fhir.r4.model.IdType;
import org.hl7.fhir.r4.model.Organization;
import org.hl7.fhir.r4.model.Parameters;
import org.hl7.fhir.r4.model.Patient;
import org.hl7.fhir.r4.model.Resource;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.EnumSource;
import org.springframework.beans.factory.annotation.Autowired;

public class PatientCoverageC4DICSearchIT extends IntegrationTestBase {
  private Bundle searchBundle(String beneSK) {
    return getFhirClient()
        .operation()
        .onInstance(new IdType("Patient", beneSK))
        .named("$generate-insurance-card")
        .withNoParameters(Parameters.class)
        .returnResourceType(Bundle.class)
        .execute();
  }

  @Autowired private BeneficiaryRepository beneficiaryRepository;

  @Test
  void patientCoverageSearchByIdPartA() {
    var response = searchBundle(BENE_ID_PART_A_ONLY);

    // Assert the response structure
    assertNotNull(response);
    assertEquals(Bundle.BundleType.COLLECTION, response.getType());
    assertFalse(response.getEntry().isEmpty());

    assertEquals(3, response.getEntry().size());
    long patientCount =
        response.getEntry().stream().filter(e -> e.getResource() instanceof Patient).count();
    long orgCount =
        response.getEntry().stream().filter(e -> e.getResource() instanceof Organization).count();
    long coverageCount =
        response.getEntry().stream().filter(e -> e.getResource() instanceof Coverage).count();

    assertEquals(1, patientCount);
    assertEquals(1, orgCount);
    assertEquals(1, coverageCount);
    assertAllResourcesHaveValidProfilesAndIds(response);
    expectFhirNormalized().toMatchSnapshot(response);
  }

  @Test
  void patientCoverageSearchByIdPartB() {
    var response = searchBundle(BENE_ID_PART_B_ONLY);

    // Assert the response structure
    assertNotNull(response);
    assertEquals(Bundle.BundleType.COLLECTION, response.getType());
    assertFalse(response.getEntry().isEmpty());

    assertEquals(3, response.getEntry().size());
    long patientCount =
        response.getEntry().stream().filter(e -> e.getResource() instanceof Patient).count();
    long orgCount =
        response.getEntry().stream().filter(e -> e.getResource() instanceof Organization).count();
    long coverageCount =
        response.getEntry().stream().filter(e -> e.getResource() instanceof Coverage).count();

    assertEquals(1, patientCount);
    assertEquals(1, orgCount);
    assertEquals(1, coverageCount);
    assertAllResourcesHaveValidProfilesAndIds(response);
    expectFhirNormalized().toMatchSnapshot(response);
  }

  @Test
  void patientCoverageSearchByIdDual() {
    var response = searchBundle(BENE_ID_DUAL_ONLY);

    // Assert the response structure
    assertNotNull(response);
    assertEquals(Bundle.BundleType.COLLECTION, response.getType());
    assertFalse(response.getEntry().isEmpty());

    assertEquals(3, response.getEntry().size());
    long patientCount =
        response.getEntry().stream().filter(e -> e.getResource() instanceof Patient).count();
    long orgCount =
        response.getEntry().stream().filter(e -> e.getResource() instanceof Organization).count();
    long coverageCount =
        response.getEntry().stream().filter(e -> e.getResource() instanceof Coverage).count();

    assertEquals(1, patientCount);
    assertEquals(1, orgCount);
    assertEquals(1, coverageCount);
    assertAllResourcesHaveValidProfilesAndIds(response);
    expectFhirNormalized().toMatchSnapshot(response);
  }

  @Test
  void patientCoverageSearchAllParts() {
    var response = searchBundle(BENE_ID_ALL_PARTS_WITH_XREF);

    // Assert the response structure
    assertNotNull(response);
    assertEquals(Bundle.BundleType.COLLECTION, response.getType());
    assertFalse(response.getEntry().isEmpty());
    assertEquals(5, response.getEntry().size());

    long patientCount =
        response.getEntry().stream().filter(e -> e.getResource() instanceof Patient).count();
    long orgCount =
        response.getEntry().stream().filter(e -> e.getResource() instanceof Organization).count();
    long coverageCount =
        response.getEntry().stream().filter(e -> e.getResource() instanceof Coverage).count();

    assertEquals(1, patientCount);
    assertEquals(1, orgCount);
    assertEquals(3, coverageCount);
    assertAllResourcesHaveValidProfilesAndIds(response);
    expectFhirNormalized().toMatchSnapshot(response);
  }

  private void assertAllResourcesHaveValidProfilesAndIds(Bundle bundle) {
    for (Bundle.BundleEntryComponent entry : bundle.getEntry()) {
      Resource resource = entry.getResource();
      assertTrue(isValidUuidUrn(resource.getId()));

      String expectedProfile =
          switch (resource) {
            case Patient p -> SystemUrls.PROFILE_C4DIC_PATIENT;
            case Organization o -> SystemUrls.PROFILE_C4DIC_ORGANIZATION;
            case Coverage c -> SystemUrls.PROFILE_C4DIC_COVERAGE;
            default -> fail("Unexpected resource type: " + resource.getClass().getSimpleName());
          };

      assertEquals(expectedProfile, resource.getMeta().getProfile().getFirst().asStringValue());
    }
  }

  private boolean isValidUuidUrn(String id) {
    if (!id.startsWith("urn:uuid:")) {
      return false;
    }
    String uuid = id.substring("urn:uuid:".length());
    return uuid.matches("[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}");
  }

  @ParameterizedTest
  @EnumSource(SearchStyleEnum.class)
  void patientSearchByIdEmpty(SearchStyleEnum searchStyle) {
    var patientBundle = searchBundle("999");
    assertEquals(0, patientBundle.getEntry().size());
  }
}
