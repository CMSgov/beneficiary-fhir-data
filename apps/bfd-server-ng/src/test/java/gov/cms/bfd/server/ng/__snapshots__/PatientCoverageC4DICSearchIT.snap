gov.cms.bfd.server.ng.PatientCoverageC4DICSearchIT.patientCoverageSearchAllParts=[
{
  "entry" : [ {
    "fullUrl" : "{uuid}",
    "resource" : {
      "birthDate" : "1967-03-19",
      "communication" : [ {
        "language" : {
          "coding" : [ {
            "code" : "es",
            "system" : "urn:ietf:bcp:47"
          } ]
        }
      } ],
      "extension" : [ {
        "url" : "http://hl7.org/fhir/us/core/StructureDefinition/us-core-sex",
        "valueCode" : "248152002"
      }, {
        "extension" : [ {
          "url" : "ombCategory",
          "valueCoding" : {
            "code" : "2054-5",
            "display" : "Black or African American",
            "system" : "urn:oid:2.16.840.1.113883.6.238"
          }
        }, {
          "url" : "text",
          "valueCode" : "Black or African American"
        } ],
        "url" : "http://hl7.org/fhir/us/core/StructureDefinition/us-core-race"
      } ],
      "gender" : "female",
      "meta" : {
        "lastUpdated" : "9999-12-31T00:00:00.000+00:00",
        "profile" : [ "http://hl7.org/fhir/us/insurance-card/StructureDefinition/C4DIC-Patient" ]
      },
      "name" : [ {
        "family" : "Erdapfel",
        "given" : [ "Bluey", "Caroline" ]
      } ],
      "resourceType" : "Patient"
    }
  }, {
    "fullUrl" : "{uuid}",
    "resource" : {
      "active" : true,
      "contact" : [ {
        "purpose" : {
          "coding" : [ {
            "code" : "PATINF",
            "display" : "Patient",
            "system" : "http://terminology.hl7.org/CodeSystem/contactentity-type"
          } ]
        },
        "telecom" : [ {
          "system" : "phone",
          "value" : "1-800-633-4227"
        }, {
          "system" : "phone",
          "value" : "TTY: 1-877-486-2048"
        }, {
          "system" : "url",
          "value" : "www.medicare.gov"
        } ]
      } ],
      "meta" : {
        "profile" : [ "http://hl7.org/fhir/us/insurance-card/StructureDefinition/C4DIC-Organization" ]
      },
      "name" : "Centers for Medicare and Medicaid Services",
      "resourceType" : "Organization"
    }
  }, {
    "fullUrl" : "{uuid}",
    "resource" : {
      "beneficiary" : {
        "reference" : "Patient/{uuid}"
      },
      "class" : [ {
        "type" : {
          "coding" : [ {
            "code" : "plan",
            "system" : "http://terminology.hl7.org/CodeSystem/coverage-class"
          } ]
        },
        "value" : "Part A"
      } ],
      "extension" : [ {
        "url" : "https://bluebutton.cms.gov/fhir/StructureDefinition/BENE-BUYIN-CD",
        "valueAnnotation" : {
          "text" : "You may be asked to show this card when you get health care services. Only give your personal Medicare information to health care providers, or people you trust who work with Medicare on your behalf. WARNING: Intentionally misusing this card may be considered fraud and/or other violation of federal law and is punishable by law.\n\nEs posible que le pidan que muestre esta tarjeta cuando reciba servicios de cuidado médico. Solamente dé su información personal de Medicare a los proveedores de salud, sus aseguradores o personas de su confianza que trabajan con Medicare en su nombre. ¡ADVERTENCIA! El mal uso intencional de esta tarjeta puede ser considerado como fraude y/u otra violación de la ley federal y es sancionada por la ley."
        }
      }, {
        "url" : "https://bluebutton.cms.gov/fhir/StructureDefinition/BENE-MDCR-ENTLMT-RSN-CD",
        "valueCoding" : {
          "code" : "0",
          "system" : "https://bluebutton.cms.gov/fhir/CodeSystem/BENE-MDCR-ENTLMT-RSN-CD"
        }
      } ],
      "identifier" : [ {
        "assigner" : {
          "reference" : "Organization/{uuid}"
        },
        "system" : "http://hl7.org/fhir/sid/us-mbi",
        "type" : {
          "coding" : [ {
            "code" : "MB",
            "system" : "http://terminology.hl7.org/CodeSystem/v2-0203"
          } ]
        },
        "value" : "8Z73WV0QC20"
      } ],
      "meta" : {
        "lastUpdated" : "9999-12-31T00:00:00.000+00:00",
        "profile" : [ "http://hl7.org/fhir/us/insurance-card/StructureDefinition/C4DIC-Coverage" ]
      },
      "payor" : [ {
        "reference" : "Organization/{uuid}"
      } ],
      "period" : {
        "end" : "2099-12-31",
        "start" : "2024-01-01"
      },
      "relationship" : {
        "coding" : [ {
          "code" : "self",
          "display" : "Self",
          "system" : "http://terminology.hl7.org/CodeSystem/subscriber-relationship"
        } ]
      },
      "resourceType" : "Coverage",
      "status" : "active",
      "subscriber" : {
        "reference" : "Patient/{uuid}"
      },
      "subscriberId" : "8Z73WV0QC20",
      "type" : {
        "coding" : [ {
          "code" : "1",
          "display" : "MEDICARE",
          "system" : "https://nahdo.org/sopt"
        } ]
      }
    }
  }, {
    "fullUrl" : "{uuid}",
    "resource" : {
      "beneficiary" : {
        "reference" : "Patient/{uuid}"
      },
      "class" : [ {
        "type" : {
          "coding" : [ {
            "code" : "plan",
            "system" : "http://terminology.hl7.org/CodeSystem/coverage-class"
          } ]
        },
        "value" : "Part B"
      } ],
      "extension" : [ {
        "url" : "https://bluebutton.cms.gov/fhir/StructureDefinition/BENE-BUYIN-CD",
        "valueAnnotation" : {
          "text" : "You may be asked to show this card when you get health care services. Only give your personal Medicare information to health care providers, or people you trust who work with Medicare on your behalf. WARNING: Intentionally misusing this card may be considered fraud and/or other violation of federal law and is punishable by law.\n\nEs posible que le pidan que muestre esta tarjeta cuando reciba servicios de cuidado médico. Solamente dé su información personal de Medicare a los proveedores de salud, sus aseguradores o personas de su confianza que trabajan con Medicare en su nombre. ¡ADVERTENCIA! El mal uso intencional de esta tarjeta puede ser considerado como fraude y/u otra violación de la ley federal y es sancionada por la ley."
        }
      }, {
        "url" : "https://bluebutton.cms.gov/fhir/StructureDefinition/BENE-MDCR-ENTLMT-RSN-CD",
        "valueCoding" : {
          "code" : "0",
          "system" : "https://bluebutton.cms.gov/fhir/CodeSystem/BENE-MDCR-ENTLMT-RSN-CD"
        }
      } ],
      "identifier" : [ {
        "assigner" : {
          "reference" : "Organization/{uuid}"
        },
        "system" : "http://hl7.org/fhir/sid/us-mbi",
        "type" : {
          "coding" : [ {
            "code" : "MB",
            "system" : "http://terminology.hl7.org/CodeSystem/v2-0203"
          } ]
        },
        "value" : "8Z73WV0QC20"
      } ],
      "meta" : {
        "lastUpdated" : "9999-12-31T00:00:00.000+00:00",
        "profile" : [ "http://hl7.org/fhir/us/insurance-card/StructureDefinition/C4DIC-Coverage" ]
      },
      "payor" : [ {
        "reference" : "Organization/{uuid}"
      } ],
      "period" : {
        "end" : "2099-12-31",
        "start" : "2022-01-01"
      },
      "relationship" : {
        "coding" : [ {
          "code" : "self",
          "display" : "Self",
          "system" : "http://terminology.hl7.org/CodeSystem/subscriber-relationship"
        } ]
      },
      "resourceType" : "Coverage",
      "status" : "active",
      "subscriber" : {
        "reference" : "Patient/{uuid}"
      },
      "subscriberId" : "8Z73WV0QC20",
      "type" : {
        "coding" : [ {
          "code" : "121",
          "display" : "MEDICARE FFS",
          "system" : "https://nahdo.org/sopt"
        } ]
      }
    }
  }, {
    "fullUrl" : "{uuid}",
    "resource" : {
      "beneficiary" : {
        "reference" : "Patient/{uuid}"
      },
      "class" : [ {
        "type" : {
          "coding" : [ {
            "code" : "plan",
            "system" : "http://terminology.hl7.org/CodeSystem/coverage-class"
          } ]
        },
        "value" : "Dual Medicare/Medicaid"
      } ],
      "extension" : [ {
        "url" : "https://bluebutton.cms.gov/fhir/StructureDefinition/BENE-BUYIN-CD",
        "valueAnnotation" : {
          "text" : "You may be asked to show this card when you get health care services. Only give your personal Medicare information to health care providers, or people you trust who work with Medicare on your behalf. WARNING: Intentionally misusing this card may be considered fraud and/or other violation of federal law and is punishable by law.\n\nEs posible que le pidan que muestre esta tarjeta cuando reciba servicios de cuidado médico. Solamente dé su información personal de Medicare a los proveedores de salud, sus aseguradores o personas de su confianza que trabajan con Medicare en su nombre. ¡ADVERTENCIA! El mal uso intencional de esta tarjeta puede ser considerado como fraude y/u otra violación de la ley federal y es sancionada por la ley."
        }
      } ],
      "identifier" : [ {
        "assigner" : {
          "reference" : "Organization/{uuid}"
        },
        "system" : "http://hl7.org/fhir/sid/us-mbi",
        "type" : {
          "coding" : [ {
            "code" : "MB",
            "system" : "http://terminology.hl7.org/CodeSystem/v2-0203"
          } ]
        },
        "value" : "8Z73WV0QC20"
      } ],
      "meta" : {
        "lastUpdated" : "9999-12-31T00:00:00.000+00:00",
        "profile" : [ "http://hl7.org/fhir/us/insurance-card/StructureDefinition/C4DIC-Coverage" ]
      },
      "payor" : [ {
        "reference" : "Organization/{uuid}"
      } ],
      "period" : {
        "start" : "2018-07-01"
      },
      "relationship" : {
        "coding" : [ {
          "code" : "self",
          "display" : "Self",
          "system" : "http://terminology.hl7.org/CodeSystem/subscriber-relationship"
        } ]
      },
      "resourceType" : "Coverage",
      "status" : "active",
      "subscriber" : {
        "reference" : "Patient/{uuid}"
      },
      "subscriberId" : "8Z73WV0QC20",
      "type" : {
        "coding" : [ {
          "code" : "14",
          "display" : "Dual Eligibility Medicare/Medicaid Organization",
          "system" : "https://nahdo.org/sopt"
        } ]
      }
    }
  } ],
  "meta" : {
    "lastUpdated" : "9999-12-31T00:00:00.000+00:00"
  },
  "resourceType" : "Bundle",
  "type" : "collection"
}
]


gov.cms.bfd.server.ng.PatientCoverageC4DICSearchIT.patientCoverageSearchByIdDual=[
{
  "entry" : [ {
    "fullUrl" : "{uuid}",
    "resource" : {
      "birthDate" : "1922-04-01",
      "communication" : [ {
        "language" : {
          "coding" : [ {
            "code" : "unknown",
            "system" : "urn:ietf:bcp:47"
          } ]
        }
      } ],
      "extension" : [ {
        "url" : "http://hl7.org/fhir/us/core/StructureDefinition/us-core-sex",
        "valueCode" : "248153007"
      }, {
        "extension" : [ {
          "url" : "ombCategory",
          "valueCoding" : {
            "code" : "1002-5",
            "display" : "American Indian or Alaska Native",
            "system" : "urn:oid:2.16.840.1.113883.6.238"
          }
        }, {
          "url" : "text",
          "valueCode" : "American Indian or Alaska Native"
        } ],
        "url" : "http://hl7.org/fhir/us/core/StructureDefinition/us-core-race"
      } ],
      "gender" : "male",
      "meta" : {
        "lastUpdated" : "9999-12-31T00:00:00.000+00:00",
        "profile" : [ "http://hl7.org/fhir/us/insurance-card/StructureDefinition/C4DIC-Patient" ]
      },
      "name" : [ {
        "family" : "Jones",
        "given" : [ "Abby" ]
      } ],
      "resourceType" : "Patient"
    }
  }, {
    "fullUrl" : "{uuid}",
    "resource" : {
      "active" : true,
      "contact" : [ {
        "purpose" : {
          "coding" : [ {
            "code" : "PATINF",
            "display" : "Patient",
            "system" : "http://terminology.hl7.org/CodeSystem/contactentity-type"
          } ]
        },
        "telecom" : [ {
          "system" : "phone",
          "value" : "1-800-633-4227"
        }, {
          "system" : "phone",
          "value" : "TTY: 1-877-486-2048"
        }, {
          "system" : "url",
          "value" : "www.medicare.gov"
        } ]
      } ],
      "meta" : {
        "profile" : [ "http://hl7.org/fhir/us/insurance-card/StructureDefinition/C4DIC-Organization" ]
      },
      "name" : "Centers for Medicare and Medicaid Services",
      "resourceType" : "Organization"
    }
  }, {
    "fullUrl" : "{uuid}",
    "resource" : {
      "beneficiary" : {
        "reference" : "Patient/{uuid}"
      },
      "class" : [ {
        "type" : {
          "coding" : [ {
            "code" : "plan",
            "system" : "http://terminology.hl7.org/CodeSystem/coverage-class"
          } ]
        },
        "value" : "Dual Medicare/Medicaid"
      } ],
      "extension" : [ {
        "url" : "https://bluebutton.cms.gov/fhir/StructureDefinition/BENE-BUYIN-CD",
        "valueAnnotation" : {
          "text" : "You may be asked to show this card when you get health care services. Only give your personal Medicare information to health care providers, or people you trust who work with Medicare on your behalf. WARNING: Intentionally misusing this card may be considered fraud and/or other violation of federal law and is punishable by law.\n\nEs posible que le pidan que muestre esta tarjeta cuando reciba servicios de cuidado médico. Solamente dé su información personal de Medicare a los proveedores de salud, sus aseguradores o personas de su confianza que trabajan con Medicare en su nombre. ¡ADVERTENCIA! El mal uso intencional de esta tarjeta puede ser considerado como fraude y/u otra violación de la ley federal y es sancionada por la ley."
        }
      } ],
      "identifier" : [ {
        "assigner" : {
          "reference" : "Organization/{uuid}"
        },
        "system" : "http://hl7.org/fhir/sid/us-mbi",
        "type" : {
          "coding" : [ {
            "code" : "MB",
            "system" : "http://terminology.hl7.org/CodeSystem/v2-0203"
          } ]
        },
        "value" : "1SJ2CG2UC85"
      } ],
      "meta" : {
        "lastUpdated" : "9999-12-31T00:00:00.000+00:00",
        "profile" : [ "http://hl7.org/fhir/us/insurance-card/StructureDefinition/C4DIC-Coverage" ]
      },
      "payor" : [ {
        "reference" : "Organization/{uuid}"
      } ],
      "period" : {
        "start" : "2020-04-08"
      },
      "relationship" : {
        "coding" : [ {
          "code" : "self",
          "display" : "Self",
          "system" : "http://terminology.hl7.org/CodeSystem/subscriber-relationship"
        } ]
      },
      "resourceType" : "Coverage",
      "status" : "active",
      "subscriber" : {
        "reference" : "Patient/{uuid}"
      },
      "subscriberId" : "1SJ2CG2UC85",
      "type" : {
        "coding" : [ {
          "code" : "14",
          "display" : "Dual Eligibility Medicare/Medicaid Organization",
          "system" : "https://nahdo.org/sopt"
        } ]
      }
    }
  } ],
  "meta" : {
    "lastUpdated" : "9999-12-31T00:00:00.000+00:00"
  },
  "resourceType" : "Bundle",
  "type" : "collection"
}
]


gov.cms.bfd.server.ng.PatientCoverageC4DICSearchIT.patientCoverageSearchByIdPartA=[
{
  "entry" : [ {
    "fullUrl" : "{uuid}",
    "resource" : {
      "birthDate" : "1922-04-01",
      "communication" : [ {
        "language" : {
          "coding" : [ {
            "code" : "unknown",
            "system" : "urn:ietf:bcp:47"
          } ]
        }
      } ],
      "extension" : [ {
        "url" : "http://hl7.org/fhir/us/core/StructureDefinition/us-core-sex",
        "valueCode" : "248153007"
      }, {
        "extension" : [ {
          "url" : "ombCategory",
          "valueCoding" : {
            "code" : "1002-5",
            "display" : "American Indian or Alaska Native",
            "system" : "urn:oid:2.16.840.1.113883.6.238"
          }
        }, {
          "url" : "text",
          "valueCode" : "American Indian or Alaska Native"
        } ],
        "url" : "http://hl7.org/fhir/us/core/StructureDefinition/us-core-race"
      } ],
      "gender" : "male",
      "meta" : {
        "lastUpdated" : "9999-12-31T00:00:00.000+00:00",
        "profile" : [ "http://hl7.org/fhir/us/insurance-card/StructureDefinition/C4DIC-Patient" ]
      },
      "name" : [ {
        "family" : "Jones",
        "given" : [ "Abby" ]
      } ],
      "resourceType" : "Patient"
    }
  }, {
    "fullUrl" : "{uuid}",
    "resource" : {
      "active" : true,
      "contact" : [ {
        "purpose" : {
          "coding" : [ {
            "code" : "PATINF",
            "display" : "Patient",
            "system" : "http://terminology.hl7.org/CodeSystem/contactentity-type"
          } ]
        },
        "telecom" : [ {
          "system" : "phone",
          "value" : "1-800-633-4227"
        }, {
          "system" : "phone",
          "value" : "TTY: 1-877-486-2048"
        }, {
          "system" : "url",
          "value" : "www.medicare.gov"
        } ]
      } ],
      "meta" : {
        "profile" : [ "http://hl7.org/fhir/us/insurance-card/StructureDefinition/C4DIC-Organization" ]
      },
      "name" : "Centers for Medicare and Medicaid Services",
      "resourceType" : "Organization"
    }
  }, {
    "fullUrl" : "{uuid}",
    "resource" : {
      "beneficiary" : {
        "reference" : "Patient/{uuid}"
      },
      "class" : [ {
        "type" : {
          "coding" : [ {
            "code" : "plan",
            "system" : "http://terminology.hl7.org/CodeSystem/coverage-class"
          } ]
        },
        "value" : "Part A"
      } ],
      "extension" : [ {
        "url" : "https://bluebutton.cms.gov/fhir/StructureDefinition/BENE-BUYIN-CD",
        "valueAnnotation" : {
          "text" : "You may be asked to show this card when you get health care services. Only give your personal Medicare information to health care providers, or people you trust who work with Medicare on your behalf. WARNING: Intentionally misusing this card may be considered fraud and/or other violation of federal law and is punishable by law.\n\nEs posible que le pidan que muestre esta tarjeta cuando reciba servicios de cuidado médico. Solamente dé su información personal de Medicare a los proveedores de salud, sus aseguradores o personas de su confianza que trabajan con Medicare en su nombre. ¡ADVERTENCIA! El mal uso intencional de esta tarjeta puede ser considerado como fraude y/u otra violación de la ley federal y es sancionada por la ley."
        }
      }, {
        "url" : "https://bluebutton.cms.gov/fhir/StructureDefinition/BENE-MDCR-ENTLMT-RSN-CD",
        "valueCoding" : {
          "code" : "0",
          "system" : "https://bluebutton.cms.gov/fhir/CodeSystem/BENE-MDCR-ENTLMT-RSN-CD"
        }
      } ],
      "identifier" : [ {
        "assigner" : {
          "reference" : "Organization/{uuid}"
        },
        "system" : "http://hl7.org/fhir/sid/us-mbi",
        "type" : {
          "coding" : [ {
            "code" : "MB",
            "system" : "http://terminology.hl7.org/CodeSystem/v2-0203"
          } ]
        },
        "value" : "9OJ2CG2UC85"
      } ],
      "meta" : {
        "lastUpdated" : "9999-12-31T00:00:00.000+00:00",
        "profile" : [ "http://hl7.org/fhir/us/insurance-card/StructureDefinition/C4DIC-Coverage" ]
      },
      "payor" : [ {
        "reference" : "Organization/{uuid}"
      } ],
      "period" : {
        "end" : "2099-12-31",
        "start" : "2024-01-01"
      },
      "relationship" : {
        "coding" : [ {
          "code" : "self",
          "display" : "Self",
          "system" : "http://terminology.hl7.org/CodeSystem/subscriber-relationship"
        } ]
      },
      "resourceType" : "Coverage",
      "status" : "active",
      "subscriber" : {
        "reference" : "Patient/{uuid}"
      },
      "subscriberId" : "9OJ2CG2UC85",
      "type" : {
        "coding" : [ {
          "code" : "1",
          "display" : "MEDICARE",
          "system" : "https://nahdo.org/sopt"
        } ]
      }
    }
  } ],
  "meta" : {
    "lastUpdated" : "9999-12-31T00:00:00.000+00:00"
  },
  "resourceType" : "Bundle",
  "type" : "collection"
}
]


gov.cms.bfd.server.ng.PatientCoverageC4DICSearchIT.patientCoverageSearchByIdPartB=[
{
  "entry" : [ {
    "fullUrl" : "{uuid}",
    "resource" : {
      "birthDate" : "1959-03-14",
      "communication" : [ {
        "language" : {
          "coding" : [ {
            "code" : "es",
            "system" : "urn:ietf:bcp:47"
          } ]
        }
      } ],
      "extension" : [ {
        "url" : "http://hl7.org/fhir/us/core/StructureDefinition/us-core-sex",
        "valueCode" : "248153007"
      }, {
        "extension" : [ {
          "url" : "ombCategory",
          "valueCoding" : {
            "code" : "2028-9",
            "display" : "Asian",
            "system" : "urn:oid:2.16.840.1.113883.6.238"
          }
        }, {
          "url" : "text",
          "valueCode" : "Asian"
        } ],
        "url" : "http://hl7.org/fhir/us/core/StructureDefinition/us-core-race"
      } ],
      "gender" : "male",
      "meta" : {
        "lastUpdated" : "9999-12-31T00:00:00.000+00:00",
        "profile" : [ "http://hl7.org/fhir/us/insurance-card/StructureDefinition/C4DIC-Patient" ]
      },
      "name" : [ {
        "family" : "Abby",
        "given" : [ "Abby" ]
      } ],
      "resourceType" : "Patient"
    }
  }, {
    "fullUrl" : "{uuid}",
    "resource" : {
      "active" : true,
      "contact" : [ {
        "purpose" : {
          "coding" : [ {
            "code" : "PATINF",
            "display" : "Patient",
            "system" : "http://terminology.hl7.org/CodeSystem/contactentity-type"
          } ]
        },
        "telecom" : [ {
          "system" : "phone",
          "value" : "1-800-633-4227"
        }, {
          "system" : "phone",
          "value" : "TTY: 1-877-486-2048"
        }, {
          "system" : "url",
          "value" : "www.medicare.gov"
        } ]
      } ],
      "meta" : {
        "profile" : [ "http://hl7.org/fhir/us/insurance-card/StructureDefinition/C4DIC-Organization" ]
      },
      "name" : "Centers for Medicare and Medicaid Services",
      "resourceType" : "Organization"
    }
  }, {
    "fullUrl" : "{uuid}",
    "resource" : {
      "beneficiary" : {
        "reference" : "Patient/{uuid}"
      },
      "class" : [ {
        "type" : {
          "coding" : [ {
            "code" : "plan",
            "system" : "http://terminology.hl7.org/CodeSystem/coverage-class"
          } ]
        },
        "value" : "Part B"
      } ],
      "extension" : [ {
        "url" : "https://bluebutton.cms.gov/fhir/StructureDefinition/BENE-BUYIN-CD",
        "valueAnnotation" : {
          "text" : "You may be asked to show this card when you get health care services. Only give your personal Medicare information to health care providers, or people you trust who work with Medicare on your behalf. WARNING: Intentionally misusing this card may be considered fraud and/or other violation of federal law and is punishable by law.\n\nEs posible que le pidan que muestre esta tarjeta cuando reciba servicios de cuidado médico. Solamente dé su información personal de Medicare a los proveedores de salud, sus aseguradores o personas de su confianza que trabajan con Medicare en su nombre. ¡ADVERTENCIA! El mal uso intencional de esta tarjeta puede ser considerado como fraude y/u otra violación de la ley federal y es sancionada por la ley."
        }
      }, {
        "url" : "https://bluebutton.cms.gov/fhir/StructureDefinition/BENE-MDCR-ENTLMT-RSN-CD",
        "valueCoding" : {
          "code" : "0",
          "system" : "https://bluebutton.cms.gov/fhir/CodeSystem/BENE-MDCR-ENTLMT-RSN-CD"
        }
      } ],
      "identifier" : [ {
        "assigner" : {
          "reference" : "Organization/{uuid}"
        },
        "system" : "http://hl7.org/fhir/sid/us-mbi",
        "type" : {
          "coding" : [ {
            "code" : "MB",
            "system" : "http://terminology.hl7.org/CodeSystem/v2-0203"
          } ]
        },
        "value" : "1L26RW8YW56"
      } ],
      "meta" : {
        "lastUpdated" : "9999-12-31T00:00:00.000+00:00",
        "profile" : [ "http://hl7.org/fhir/us/insurance-card/StructureDefinition/C4DIC-Coverage" ]
      },
      "payor" : [ {
        "reference" : "Organization/{uuid}"
      } ],
      "period" : {
        "end" : "2099-12-31",
        "start" : "2024-01-01"
      },
      "relationship" : {
        "coding" : [ {
          "code" : "self",
          "display" : "Self",
          "system" : "http://terminology.hl7.org/CodeSystem/subscriber-relationship"
        } ]
      },
      "resourceType" : "Coverage",
      "status" : "active",
      "subscriber" : {
        "reference" : "Patient/{uuid}"
      },
      "subscriberId" : "1L26RW8YW56",
      "type" : {
        "coding" : [ {
          "code" : "121",
          "display" : "MEDICARE FFS",
          "system" : "https://nahdo.org/sopt"
        } ]
      }
    }
  } ],
  "meta" : {
    "lastUpdated" : "9999-12-31T00:00:00.000+00:00"
  },
  "resourceType" : "Bundle",
  "type" : "collection"
}
]